plugins/resources/replication/irods_repl_rebalance.cpp

```diff
@@ -96,12 +96,13 @@ namespace {
         addInxVal(&genquery_inp_wrapped.get().sqlCondInp, COL_D_DATA_ID, (boost::format("= '%lld'") % _data_id).str().c_str());
 
         std::stringstream cond_ss;
+        cond_ss << "IN (";
         for (auto& bun : _leaf_bundles) {
             for (auto id : bun) {
-                cond_ss << "= '" << id << "' || ";
+                cond_ss << "'" << id << "',";
             }
         }
-        std::string cond_str = cond_ss.str().substr(0, cond_ss.str().size()-4);
+        const std::string cond_str = std::string(cond_ss.str().substr(0, cond_ss.str().size()-1)) + ")";
```

Comment:
Perhaps a comment for the -1 would be helpful. It's truncating the trailing comma, right?

Reply:
there is likely a more memory efficient way to do this. although, the compiler may be smart enough to elide things for us.

Reply:
the `std::string(...)` constructor calls is not needed. `substr()` returns a `std::string`

Reply:
yes on kicking the can... RESC_IDs could be 'big' so it's hard to know the actual max itself.

will remove the extra `std::string`.

Reply:
done, done